src/test/java/DeadLinksTest.java

```diff
@@ -0,0 +1,55 @@
+import com.google.gson.*;
+import java.io.*;
+import org.junit.*;
+import org.junit.runner.*;
+import org.junit.runners.*;
+
+import static org.junit.Assert.*;
+
+@RunWith(JUnit4.class)
+public class DeadLinksTest {
+
+  private final ByteArrayOutputStream actualOutput = new ByteArrayOutputStream();
+  private final ByteArrayOutputStream actualErrOutput = new ByteArrayOutputStream();
+  private String expectedOutput;
+
+  @Before
+  public void setUpStreams() {
+    System.setOut(new PrintStream(actualOutput));
+    System.setErr(new PrintStream(actualErrOutput));
+  }
+
+  @Before
+  public void readExpectedJSON() {
+    Gson gson = new GsonBuilder().setPrettyPrinting().create();
+    JsonObject jsonObject = gson.fromJson(new InputStreamReader(
+        getClass()
+            .getClassLoader()
+            .getResourceAsStream("expected_output.json")), JsonObject.class);
+
+    expectedOutput = gson.toJson(jsonObject);
+  }
+
+  @Test
+  public void MainOutputTest() {
```

Comment:
@VAndreyD the test method naming is not followed the one described at
https://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names